### ISSUE TITLE
Resizing the predictions to the original image shape

### ISSUE THREAD

**Issue by IssamLaradji** [facebookresearch/maskrcnn-benchmark#564]
## ❓ Questions and Help

In the inference code, how does `prediction.resize((image_width, image_height))` know how to deal with the padding from `SIZE_DIVISIBILITY: 32`? 

Wouldn't the predictions be slightly off?

Thanks!

**Comment by muzammil360** [comment]
@IssamLaradji, I think `SIZE_DIVISIBILITY` is a dataloader parameter i.e. no. of samples (patches from images). This is NOT  `[w,h]` size.

Look in the default config file [here](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/config/defaults.py)
`# If > 0, this enforces that each collated batch should have a size divisible`
`# by SIZE_DIVISIBILITY`
`_C.DATALOADER.SIZE_DIVISIBILITY = 0`

**Comment by TyroneLi** [comment]
I agree with@muzammil360,that parameter should be the division of the total dataset how much to process per batch,and this can be adjusted depended on your machine capaciity.

**Comment by HawaiiZeng** [comment]
@muzammil360 It should be referring to image shape `w, h`. See this thread: [https://github.com/facebookresearch/maskrcnn-benchmark/issues/419](url)
I guess the slightly off is what happens here. @IssamLaradji
